javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -603,6 +601,96 @@ public static ClientMethodMapper getInstance() {
+        // implementation of the previous AutoRest generation (Flux<ByteBuffer>, etc) will pass the body as a new
+        // instance of BinaryData, ex BinaryData.fromFlux(flux).
+        //
+        // This is done to ease the transition for SDK authors, aka when the directive to generate request bodies as
+        // BinaryData is passed there isn't a requirement right away to translate their current SDK calls to the new
+        // BinaryData format. Instead, SDK authors will be able to port, test, and validate the transition to BinaryData
+        // when appropriate.
+        if (settings.isRequestBodyGeneratedAsBinaryData()) {
+            // Loop over each ClientMethod that was generated and for each method that has a BodyParam create a new
+            // instance that uses BinaryData as the value type.
+            //
+            // This is done using an iteration with index so that the new instance can be inserted into the list of
+            // ClientMethods to be generated next to its corresponding non-BinaryData overload.
+            for (int methodIndex = 0; methodIndex < methods.size(); methodIndex++) {
+                ClientMethod method = methods.get(methodIndex);
+                List<String> validateExpressionsToRemove = new ArrayList<>();
+
+                List<ClientMethodParameter> binaryDataMethodParameters = new ArrayList<>();
+                for (int parameterIndex = 0; parameterIndex < method.getParameters().size(); parameterIndex++) {
+                    ClientMethodParameter parameter = method.getParameters().get(parameterIndex);
+                    if (parameter.getLocation() != RequestParameterLocation.BODY || parameter.getFromClient()
+                        || (method.getOnlyRequiredParameters() && !parameter.getIsRequired())) {
+                        binaryDataMethodParameters.add(parameter);
+                        continue;
+                    }
+
+                    validateExpressionsToRemove.add(parameter.getName());
+
+                    // Copy the ClientMethodParameter and change the raw and wire type to BinaryData.
+                    binaryDataMethodParameters.add(new ClientMethodParameter.Builder()
+                        .description(parameter.getDescription())
+                        .isFinal(parameter.getIsFinal())
+                        .wireType(ClassType.BinaryData)
+                        .rawType(ClassType.BinaryData)
+                        .name(parameter.getName())
+                        .isRequired(parameter.getIsRequired())
+                        .isConstant(parameter.getIsConstant())
+                        .fromClient(parameter.getFromClient())
+                        .defaultValue(parameter.getDefaultValue())
+                        .annotations(parameter.getAnnotations())
+                        .location(parameter.getLocation())
+                        .build());
```

Comment:
These could be fragile if later someone add fields to ClientMethodParameter or ClientMethod. Should we add a e.g. `Builder.from(ClientMethodParameter parameter)` in this 2 classes instead (so at least the related code will be in same class)?